cockatrice/src/window_main.cpp

```diff
@@ -328,6 +335,36 @@ void MainWindow::loginError(Response::ResponseCode r, QString reasonStr, quint32
     actConnect();
 }
 
+void MainWindow::extractInvalidUsernameMessage(QString & in, QString & out)
```

Comment:
Is there a recommended practice for this?

Using `void` signature with an out parameter seems wrong to me. Why not return the out value normally?